src/os/bluestore/BlueStore.cc

```diff
@@ -3296,9 +3296,9 @@ int BlueStore::_blob2read_to_extents2read(
     uint64_t x_offs = cur->blob_xoffset;
     while (l > 0 && ext_it != ext_end) {
 
-      assert(blob->length >= ext_pos + r_offs);
+      assert(blob->get_payload_length() >= ext_pos + r_offs);
 
-      uint64_t r_len = MIN(blob->length - ext_pos - r_offs, ext_it->length - r_offs);
+      uint64_t r_len = MIN(blob->get_payload_length() - ext_pos - r_offs, ext_it->length - r_offs);
```

Comment:
Am I reading this correctly that these length -> get_payload_length() changes are subtle bug fixes?